src/main/kotlin/com/team4099/robot2023/subsystems/groundintake/GroundIntake.kt

```diff
@@ -197,32 +202,23 @@ class GroundIntake(private val io: GroundIntakeIO) : SubsystemBase() {
 
     var nextState = currentState
     when (currentState) {
-      GroundIntakeConstants.ArmStates.UNINITIALIZED -> {
+      GroundIntakeState.UNINITIALIZED -> {
         // Outputs
         // No designated output functionality because targeting position will take care of it next
         // loop cycle
 
         // Transitions
-        if (requestedState == GroundIntakeConstants.ArmStates.UNINITIALIZED) {
-          requestedState = GroundIntakeConstants.ArmStates.TARGETING_POSITION
-          nextState = GroundIntakeConstants.ArmStates.TARGETING_POSITION
-        } else {
-          nextState = requestedState
-        }
+        nextState = fromRequestToState(requestedState)
       }
-      GroundIntakeConstants.ArmStates.OPEN_LOOP -> {
+      GroundIntakeState.OPEN_LOOP_REQUEST -> {
         // Outputs
         setArmVoltage(armVoltageTarget)
         setRollerVoltage(rollerVoltageTarget)
 
         // Transitions
-        if (requestedState == GroundIntakeConstants.ArmStates.UNINITIALIZED) {
-          requestedState = GroundIntakeConstants.ArmStates.OPEN_LOOP
-        } else {
-          nextState = requestedState
-        }
+        nextState = fromRequestToState(requestedState)
```

Comment:
this is fine here but in superstructure, this function should not exist. I guess the way I would put it is, you will not be able to get the next state just from the requested state. You could break all of the transitions out into a function, and make this `fromRequestToState(currentState: State, request: Request)` or something like this, but I don't think that is actually worth it. Generally it makes sense to have the logic just in here.

Reply:
we can still add the logic tho. the only reason i included this is because if we don't have any logic necessary and just want to map the requested state to the next state we need something like this (in all the possible current states). we could still add logic there but rn isn't any bc request can never be uninitialized